enterprise/backend/src/metabase_enterprise/advanced_permissions/common.clj

```diff
@@ -64,8 +64,8 @@
     :else
     (filter
      (fn [{db-id :db_id schema :schema}]
-       (perms/set-has-full-permissions? @api/*current-user-permissions-set*
-                                        (perms/feature-perms-path :data-model :all db-id schema)))
+       (perms/set-has-partial-permissions? @api/*current-user-permissions-set*
```

Comment:
This fixes a bug where the user could have data model permissions for a table inside a schema, but the `api/database/:id/schemas` endpoint might not return the schema because they could only have partial permissions, not full permissions. See this test 
https://github.com/metabase/metabase/pull/30352/files#diff-6b7ca992d87c53539a7526cd61b5570fc37d142132b94e622980719cbe5bcf9fR272